graph/graph.go

```diff
@@ -141,14 +142,92 @@ func (graph *Graph) restore() error {
 		}
 	}
 
-	baseIds, err := graph.restoreBaseImages()
+	graph.idIndex = truncindex.NewTruncIndex(ids)
+	logrus.Debugf("Restored %d elements", len(ids))
+	return nil
+}
+
+func (graph *Graph) RestoreCustomImages(ts *TagStore) error {
```

Comment:
This should still be in the graph layer. In Go, we generally avoid cycles by interface definition. We can define a `Tagger` interface in `graphdriver/windows` or `graphdriver`, which `TagStore` implements.

```go
// in graphdriver or graphdriver/windows
type Tagger interface {
    Tag(repoName, tag, imageName string, force bool) error
}

// in graphdriver or graphdriver/windows
type ImageRestorer {
    RestoreCustomImages(tagger Tagger) error
}

// in graphdriver/windows package
func (d *Driver) RestoreCustomImages(tagger Tagger) error
```

This allows us to inject a tagstore with depending on the `graph` package. The responsibility is kept within the driver, but can rely on requiring tagging behavior.

Wherever we may call this would like like this:

```
if r, ok := driver.(ImageRestorer); ok {
    if err := r.RestoreCustomImages(myTagStore); err != nil { // ... }
}
```

From my understanding, this is used as part of the the startup, so that is where we'd do this test.